source/extensions/filters/udp/udp_proxy/session_filters/dynamic_forward_proxy/proxy_filter.h

```diff
@@ -85,10 +85,14 @@ class ProxyFilter
   void onLoadDnsCacheComplete(
       const Extensions::Common::DynamicForwardProxy::DnsHostInfoSharedPtr&) override;
 
+  bool bufferEnabled() const { return buffer_enabled_; };
```

Comment:
Recommend calling it "sessionBufferEnabled" to clarify this is session level status